### ISSUE TITLE
[Feature Request] Background image size selection

### ISSUE THREAD

**Issue by zapappa** [pojome/elementor#3722]
Using WordPress 4.9.4 running GeneratePress with Elementor 1.9.7

If I add an image to a page without Elementor, by using Add Media, I see the ATTACHMENT DISPLAY SETTINGS that allows me to specify one of the four available sizes. 

If I add an image widget to a page in Elementor and then click Add Media I see the ATTACHMENT DISPLAY SETTINGS.  The widget also shows the size settings.

However, if I add an image as background (eg background image for a section) the ATTACHMENT DISPLAY SETTINGS is not shown in the Insert Media display and nor can the photo size be selected in the element settings.  There is a size selection in the element settings but that is for how the image is to be displayed (cover,contain etc) and not the image size that should be selected.

So, if I have a small area for which I want to set a background image,  when I choose the image it automatically selects the full size original image.  This could mean a 5000x3000 pixel image getting selected when I only need to fill a 300x200 pixel area.

Could you just allow the Insert Media display to include the ATTACHMENT DISPLAY SETTINGS image sizes?

**Comment by bainternet** [comment]
Currently, this feature is not supported, but we will look into it in the future if we see a large demand for it.

Thanks for the feedback.

**Comment by ShadowXVII** [comment]
I swear Background Size used to be an option at the Section level. I recall setting it, but now can't find it since updating to the latest 2.1.1. i.e.. Centre Centre, Centre Top, Centre Bottom, etc.

The sizing feature would also be useful -- I'm using ACF at the moment to store the image, so can specify the size which is a workaround for this.

**Comment by benhohner** [comment]
@bainternet Can you please reopen this ticket? I currently have to request resized images from our designers when it should be as easy as WP autoresizing it and me selecting the size I want from a dropdown.

**Comment by bentalgad** [comment]
+1 This is so needed! our sites load bigger images then we need and makes our sites slow!

**Comment by ShadowXVII** [comment]
The ACF workaround is a pain. I've ended up resizing all my full images so they're smaller. Not ideal.

WordPress supports image sizing, so it shouldn't be hard to implement.

**Comment by damienrave** [comment]
+1 for this one. Downloading full size background images for small containers is a waste of bandwidth. Users upload 4MB images which we display at 400x400, this is a performance killer... Please !

**Comment by bentalgad** [comment]
Also theres need to be a background image lazy load option!

**Comment by TheMangoTrain** [comment]
+1 for this feature.

To manually resize and upload different image sizes makes no sense when WP already has a mature system for automatically making size variants.

Here's a workaround I use, which is to render the background with CSS, instead of Elementor controls, and to inject my own custom URL (pointing to a WP image size variant).

1 - Using the standard Elementor background controls, setup your background image as you'd like, with the original full resolution image.
2 - View the page fully rendered in a new browser tab, ie either preview or publish.
3 - With your browser's inspector, find the background image CSS and copy the relevant CSS lines (they'll be in one group) with the id/class. For example, in my recent case it looked like this:

```
.elementor-1421 .elementor-element.elementor-element-934176d > .elementor-element-populated {
    background-image: url(https://www.mywebsite.com/wp-content/uploads/2018/12/photo.jpg);
    background-position: center center;
    background-repeat: no-repeat;
    background-size: cover;
    box-shadow: 0px 10px 27px -17px rgba(0,0,0,.5);
    transition: background .3s,border .3s,border-radius .3s,box-shadow .3s;
    padding: 4em 0 4em 1em;
}
```

4 - Go back to the Elementor editor, clear the background image. Paste the CSS code in Advanced/CSS.
5 - At this point you're just manually rendering the very same background image, just without using Elementor's background options. You can save and preview again to confirm things are working.
6 - Now that you're manually rendering the background image, you can paste any image URL you want. Find the URL to the WP image size you want, and paste it into the CSS. Ultimately, the image filename will be the same, but just with a resolution defined in the end, for example:
https://www.mywebsite.com/wp-content/uploads/2018/12/photo-1024x543.jpg

**How to get the WP image URL:**
There's a variety of ways to get this, but since we're in Elementor and already in the cycle of save/preview/inspect, you may as well temporarily add the original high res photo as a plain image widget in Elementor, select a custom size, save/preview and use your inspector to get the proper URL for the WP-scaled image.

The above is not difficult, but replicated X times it's a pain in the ass, especially if images get changed in the future.

**Comment by coolwebs** [comment]
+1 from me. Surprised this feature hasn't already been implemented in Elementor.

**Comment by black-eye** [comment]
+1

**Comment by erfan-ilyas** [comment]
+1

**Comment by feedyourmind** [comment]
@ShadowXVII you should not have closed this. 
I think there is some confusion here.  The background image size you are referring to is not what this ticket is about! 

When an image is uploaded to the media library, wordpress automatically creates different versions of the image, with different resolutions,  as defined in the Media Settings  ( /wp-admin/options-media.php ).

This means that if you upload an image of 8Megabyte size, it will automatically create different verions of the huge image of much smaller footprint. When you use the image, you can typicaly select the "image size (like thumbnail, medium, large, full) and so on. 

This is extremely handy, because without having to spend effort to do the conversion of the original full image into smaller sizes, you have the smaller sizes at different resolutions to use them.

This is not supported yet for background images. When changing the size value, it is always the "full resolution" image used and what changes is only the css width and height, which is effectively like a zoom. What we need is to be able to choose one of the lower resolution images wordpress automatically creates for us. 

 It is a trivial change and it would make our lives much easier. 
+1 for this feature. 
This is still missing. 
Please reopen this issue.

**Comment by ShadowXVII** [comment]
@feedyourmind , I didn't close it -- that was @bainternet 

I've been told previously that a new issue is the way to draw attention to an older issue... though I'm not sure why they don't just re-open. More tickets seems like a pain.

I'm not sure why it's still being ignored -- it seems a pretty logical setting, and other Elementor widgets have this control.

**Comment by TheMangoTrain** [comment]
@feedyourmind Yea this is an issue not specifically addressed by the Pro 2.4 update. There's still no integration with WordPress-generated-sizes for background images in Elementor.

The Pro 2.4 update does have lots of nice responsive controls for background images. This includes being able to specify different Media images per breakpoint, which is at least something, but requires a different Media file entry for each breakpoint. So for to achieve a smaller mobile background image filesize, you'd have to manually resize it and re-upload as a new Media file.

Ideally there would be a new option labeled "Source Size", with responsive controls, and in it you select the WordPress-generated-size you'd like to use for the image you selected. Simple as that.

On a side note, there is a current option labeled "Size" with options "Default, Auto, Cover, Contain, Custom". This option invokes the official css rule 'background-size', so it's understandable that one might just label it "Size" without a second thought. But in a high-level WYSIWYG interface like Elementor, with non-techies using it, the label would be much better as something like "Sizing Method". This would help clear up confusion, with or without a "Source Size" option described above.

Ultimately, the best thing would be to make use of the CSS property `image-set`, however browser support is low. It's the future, anyways. See [my point #2 here](https://github.com/pojome/elementor/issues/2355#issuecomment-447665138).

**Comment by feedyourmind** [comment]
@ShadowXVII my apologies, you are right, I should have said: 
@bainternet you should not have closed this. This feature is a must and it should be trivial to implement, given it's already supported in most other image-related controls. 

Furthermore, I should say that i agree with everything @TheMangoTrain said. Actually not sure what the label should say, it could be possibly called media-size .

Also, yes, the html generated should have image-set included (corresonding thing of <img srcset=..> for bkground images)

**Comment by timsamuelsson** [comment]
The 2.4 update is a step in the right direction but its still a pain in the but to have to upload every image three times to make it optimized for the diffrent viewport sizes.

A better option would be to autoset the background image from an srcset. Then you could pic the correct image to display with JS. This technique could be used together with multiple div's to be able to still choos image for every breakpoint.

Example from lazysizes:
`<div class="lazyload" data-bgset="image-200.jpg 200w, image-300.jpg 300w, image-400.jpg 400w" data-sizes="auto">
`
Source: https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/bgset

**Comment by Micemade** [comment]
+1

**Comment by Perchpole** [comment]
I use a lot of custom image sizes. It's impotant to crop large photos for use as hero images and mastheads. Right now, however, Elementor will not allow me to use the cropped images as backgrounds - which completely defeats the object of the exercise!

This needs to be fixed.

+1

**Comment by blueglassdigital** [comment]
+1

Isnt its quite a large demand on this now? will you eventually add this feature? :)
**It's a pain to crop images manually...**

**Comment by theamazingaustin** [comment]
+1

**Comment by musback** [comment]
Big +1. I use a 'featured image' header in a column-background throughout the site. So I'm kind of stuck here with big images in small placeholders... This makes my pages at least 100kb heavier than necessary. Hope you guys's's can implement this. Regards, a very happy Elementor Pro user.

**Comment by Jordidriesen** [comment]
Another +1, responsive images need to be a thing!

**Comment by MHitMaN** [comment]
+1

**Comment by mostafasoufi** [comment]
+1

**Comment by mrmolaei** [comment]
+1

**Comment by erfan-ilyas** [comment]
+1 Need this feature!

**Comment by twhaas** [comment]
+1 This really needs to be added! In both sections and the slider having background images of over 2MB is a huge pain, as my clients often upload full sized images

**Comment by benjaminief** [comment]
Hello,
I have same problem. I do it manually in the php file. 
here : elementor-pro/modules/dynamic-tags/tags/post-featured-image.php
on line 36; 
replace :
 'url' => wp_get_attachment_image_src( $thumbnail_id, 'full' )[0],
by
'url' => wp_get_attachment_image_src( $thumbnail_id, 'large' )[0],

**Comment by jonathanfunk** [comment]
+1

**Comment by romus1989** [comment]
+1

**Comment by rathert** [comment]
+1

**Comment by KingGrizzly** [comment]
+1

My client site is getting crushed on pae speed because it is delivering large images for all background images. 

I find it necessary to use many bg images in columns to help me consistently control image height in layouts using the background-size: cover setting.

Thanks.

**Comment by R2wix9mtdI** [comment]
I think this issue has been left long enough and needs to see a solution implemented. I have the pro version and this along is killing my SEO. It's no longer a nice to have. C'mon team help us all out

**Comment by mjb2000** [comment]
+1 from me

I have been trying to find how to do this in Elementor for the past 30 minutes.. I figured it had to be there because so many other features are so well implemented, but this is certainly something big that is missing.

Particularly for me, I am choosing a Product Image in WooCommerce and then using the "dynamic" feature to display the image in a "Single Product" template page. The product image selected is the original file, so it would be great if I could choose a different image size from the predefined Wordpress image variations for each of the responsive layouts.

**Comment by haet** [comment]
I've got the same problem. I found a solution to use a predefined image size for all background images. This is not a solution but at least better then using full image size and of course you can replace "large" by any size you prefer:
`
// limit background image sizes to size "large" instead of "full"
add_action(	'elementor/element/parse_css', function( $post_css, $element ) {
    /**
    * @var \Elementor\Post_CSS_File $post_css
    * @var \Elementor\Element_Base  $element
    */
    $rules = $post_css->get_stylesheet()->get_rules( );
    if( is_array( $rules ) && count( $rules ) ){
        foreach ($rules as $device_key => $device_rules ) {
            if( is_array( $device_rules ) && count( $device_rules ) ){
                foreach ($device_rules as $selector => $styles ) {
                    if( is_array( $styles ) && count( $styles ) ){
                        foreach ($styles as $attr => $val ) {
                            if( $attr == 'background-image' ){
                                $url = str_replace('url(','', $val);
                                $url = str_replace(')','',$url);
                                $url = str_replace("'",'',$url);
                                $url = str_replace('"','',$url);
                                $image_id = attachment_url_to_postid( $url );
                                if( $image_id ){
                                    $image = wp_get_attachment_image_src( $image_id, 'large' );
                                    $post_css->get_stylesheet()->add_rules(
                                        $selector, ['background-image' => 'url(' . $image[0] . ')']
                                    );
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}, 10, 2 );`

**Comment by black-eye** [comment]
@bainternet, please consider reopening the issue. The demand showed here seems quite large, at least to me.

**Comment by KingGrizzly** [comment]
With the release of the background images slideshow feature, I'd say the need becomes even more acute. Image 5-7 background images loading in at full size.  

Expecting clients to load optimally sized images for 3 device sizes isn't going to happen. If possible, a way to choose a preferred image size per device size would solve the challenge. Thanks!

**Comment by ShadowXVII** [comment]
This issue was closed -- it was reopened as #6778

Please thumbs up the first post to help it move up the backlog.

**Comment by John-Greg** [comment]
Just spent hours on this today, looking for a way to insert WordPress's already existing scaled images.  If there's 20 already existing images, why would we want to edit/scale/generate yet another image?  All we need is that little dropdown list available of already existing scaled images---in any situation where an image is to be inserted from the Media Library.

**Comment by bentalgad** [comment]
@John-Greg  Please add a comment [here](https://elementor.com/blog/4-million-winning-feature/)

**Comment by R2wix9mtdI** [comment]
@John-Greg Absolutelt. I recently voted on the feature request and none of them were as important as this. Site speed should be a number1 priority, so for the sake of a drop-down - please just do it. Another essential feature is infinite scroll for posts. Please, please focus on fundamentals and then ad the 'fancy stuff'. Don't get me wrong - Elementor Pro is No1 for me!

Check this response - haven't tried it yet, but it seemed a good interim:

https://github.com/elementor/elementor/issues/3722#issuecomment-529549454

**Comment by bentalgad** [comment]
I tried to leave a comment on this new announcement about this stuff, but they didn't approve the comment :) i wonder why? :)))

**Comment by maxlatitude** [comment]
+1. This feature is much needed. One option for some, The [Dynamic.ooo](url) 'Dynamic Content for Elementor' package has a featured block with background image capabilities together with the option to choose the size of the background image.

<img width="455" alt="Dynamic Content Featured Image Block" src="https://user-images.githubusercontent.com/59975293/76632808-260a8480-651a-11ea-9c5a-00489614780c.png">

**Comment by PatrickD1985** [comment]
Such a shame to see that this is still a thing in 2020. It's a core WP feature which can be easily made accessible and should be available everywhere you can select an attachment.

Because customers will now get a full size selection regardless which could mean they select a file of a couple of MB;'s big which is far from optimal when it comes to pageload.

Please Elementor team make this something which is always available when working with an attachment.

**Comment by mjb2000** [comment]
Hi @haet 

Thanks for you post a while back where you showed how we can use your code snippet to only use -large images.

However, I experienced a big problem when using different images for responsive layouts...

Using your code, when a new css .selector is added Elementor doesn't seem to care about the particular device-type that was currently being worked on, and therefore this new CSS is added outside of any media declarations. This means that images intended for use as tablet or phones end up being used on the desktop (no media declaration) version.

Any idea how the code could be updated to fix this? It's all a little beyond me.

**Comment by epoplive** [comment]
So I've figured out how to do this:
![image](https://user-images.githubusercontent.com/727892/84079662-cd6b4c00-a98f-11ea-880b-6a24775601f0.png)

You'll notice I'm loading a much smaller version of the image via a css rule.  Ignore the sizes and everything, this was just a proof of concept.  The main issue is to be able to do this you have to make a copy of one of the elementor files because alot of it's access is marked private instead of protected so you are unable to just extend it.  I'd appreciate it if you'd like to see features like this in elementor that you go show support on the issue I created #11564 which would enable this without duplicating any of their code.

**Comment by qantumthemes** [comment]
Just purchased Elementor Pro for a landing page and got astonished by the missing of this simple setting. Come on guys it really doesn't make sense, if you already uploaded a picture, that you need to go back to photoshop to just make another size, as it's already in the Wp uploads.

Also is astonishing that there is no feedback after so many +1 on this request.

I mean, it would take about 1 hour to add the feature via a plugin, why not adding it at least on Elementor PRO?

**Comment by epoplive** [comment]
> Just purchased Elementor Pro for a landing page and got astonished by the missing of this simple setting. Come on guys it really doesn't make sense, if you already uploaded a picture, that you need to go back to photoshop to just make another size, as it's already in the Wp uploads.
> 
> Also is astonishing that there is no feedback after so many +1 on this request.
> 
> I mean, it would take about 1 hour to add the feature via a plugin, why not adding it at least on Elementor PRO?

It definitely takes a lot more than an hour, largely due to bad documentation and very poor implementations of many things in the plugins.  I've been been able to accomplish this, but have ended up changing the method by which i'm accomplishing it a few times to try and figure out the best way to handle it.  I'm planning to post more info on how to do it once I finalize a few more things.  Unfortunately backgrounds are handled differently in different widgets, and nothing is anywhere near as easy as it should be.

**Comment by warudin** [comment]
Selecting the image size for background-images should be an option in Elementor. For loading times of pages it can make a huge difference if the image is being loaded in it's full size, or a smaller version.

**Comment by epoplive** [comment]
> Selecting the image size for background-images should be an option in Elementor. For loading times of pages it can make a huge difference if the image is being loaded in it's full size, or a smaller version.

At this point we should have the option to not even use a background image, and use a regular image with object-fit instead as we could use native lazy-loading.  Only ie11 doesn't support it, and I believe there is a polyfill.

**Comment by webthread** [comment]
Why is this request closed? Without these kinds of very basic features things, it makes websites ridiculous to optimise. Like seriously, why would a simple bug like this be ignored?!

**Comment by ShadowXVII** [comment]
> This issue was closed -- it was reopened as #6778
> 
> Please thumbs up the first post to help it move up the backlog.

^

**Comment by BryanRiveraLivia** [comment]
Hello guys, greetings from Peru!
Taking advantage of a bit of PHP knowledge I was able to do this using 2 plugins:
1. Woody Snipets
2. Ele Custom Skin PRO

Instructions:
1. We enter Woody Snippets and create a code snippet in PHP, this code must bring the thumbnail url.
![image](https://user-images.githubusercontent.com/10824372/86498800-132bf200-bd4d-11ea-9c07-66087dec424e.png)
`$thumb_id = get_post_thumbnail_id();
$thumb_url = wp_get_attachment_image_src($thumb_id,'thumbnail-size', true);
$featured_img_url = get_the_post_thumbnail_url(get_the_ID(),'thumbnail'); 
?>
<div style="    width: 100%;
    height: 120px;
    background-image: url(<?=$featured_img_url;?>);
    background-size: cover;
    background-repeat: no-repeat;
    background-position: top center;
">funciona</div>`

2. We enter Templates> Theme buildes> Loop, then we created a custom loop for our posts

3. Replace background image for the shortcode created in step 1.
![image](https://user-images.githubusercontent.com/10824372/86498652-618cc100-bd4c-11ea-9dae-155d01b0024b.png)

4. Result:
![image](https://user-images.githubusercontent.com/10824372/86498824-3656a180-bd4d-11ea-9af1-2e9d66b09f0a.png)

** You can custom the width o height in our CSS panel of Elementor using !important
![image](https://user-images.githubusercontent.com/10824372/86498863-69993080-bd4d-11ea-8509-5467f3967881.png)

**Comment by epoplive** [comment]
> Hello guys, greetings from Peru!
> Taking advantage of a bit of PHP knowledge I was able to do this using 2 plugins:
> 
> 1. Woody Snipets
> 2. Ele Custom Skin PRO
> 
> Instructions:
> 
> 1. We enter Woody Snippets and create a code snippet in PHP, this code must bring the thumbnail url.
>    ![image](https://user-images.githubusercontent.com/10824372/86498800-132bf200-bd4d-11ea-9c07-66087dec424e.png)
> 
> ```
> $thumb_id = get_post_thumbnail_id();
> $thumb_url = wp_get_attachment_image_src($thumb_id,'thumbnail-size', true);
> $featured_img_url = get_the_post_thumbnail_url(get_the_ID(),'thumbnail'); 
> ?>
> <div style="    width: 100%;
>     height: 120px;
>     background-image: url(<?=$featured_img_url;?>);
>     background-size: cover;
>     background-repeat: no-repeat;
>     background-position: top center;
> ">funciona</div>
> ```
> 
> 1. We enter Templates> Theme buildes> Loop, then we created a custom loop for our posts
> 2. Replace background image for the shortcode created in step 1.
>    ![image](https://user-images.githubusercontent.com/10824372/86498652-618cc100-bd4c-11ea-9dae-155d01b0024b.png)
> 3. Result:
>    ![image](https://user-images.githubusercontent.com/10824372/86498824-3656a180-bd4d-11ea-9af1-2e9d66b09f0a.png)
> 
> ** You can custom the width o height in our CSS panel of Elementor using !important
> ![image](https://user-images.githubusercontent.com/10824372/86498863-69993080-bd4d-11ea-8509-5467f3967881.png)

I posted a better solution here https://github.com/elementor/elementor/issues/6778

It lazy loads the background images and loads source sets for them so it will automatically use a more optimized version depending on screen size.

**Comment by numanagit** [comment]
Will this feature be ingnored? Two years to add a basic Wordpress core feature? Makes no sense

**Comment by ztpinar** [comment]
+1 This needs to be a thing

**Comment by pixelicous** [comment]
also in need for this..

**Comment by thomasch-easting** [comment]
We're probably still asking for this when Elementor announced they've added hyperspeed for version 6.0

**Comment by pixelicous** [comment]
> Dynamic.ooo

I found another solution for a basic feature missing in elementor pro, making the whole column as a link. installed the plugin, poof problem solved.. i agree with all said above, this could all be solved in a day's PR, instead the topic is closed

**Comment by redpik** [comment]
+1
I also need it.

**Comment by waynedavidharris** [comment]
+1 !
Elementor is a great product. But it has astounded me how long it has taken for some basic bread and butter core functionality. Global colours and fonts have only just been released after what 2-3 years - which for me is an essential basic requirement that should've been there out of the gate. And this issue of not being able to choose a crop size for a background image is another core feature crazily omitted and worse still ignored.
It is a shame that Elementor marketing seems to focus on lay-person candy floss and not catering or listening to more serious developers..
I am going to look into using Dynamic Tags to create a custom background image that allows setting the crop size. Hours of research and messing around for something that could be added by the Elementor team in an hour at the core codebase level. 
Its a weird world.

**Comment by black-eye** [comment]
Everyone adding their thumb ups here, please do so in https://github.com/elementor/elementor/issues/6778#issue-comment-box as well. This issue is closed, your voice will get more attention in the open one (at least you can hope so).

**Comment by WildPress** [comment]
```/**
 * Elementor featured image with custom size
 */
add_action('elementor/dynamic_tags/register_tags', function ($dynamic_tags) {
    class Featured_Image_Custom_Size extends \ElementorPro\Modules\DynamicTags\Tags\Base\Data_Tag
    {
        public function get_name()
        {
            return 'post-featured-image-custom-size';
        }

        public function get_group()
        {
            return \ElementorPro\Modules\DynamicTags\Module::POST_GROUP;
        }

        public function get_categories()
        {
            return [Module::IMAGE_CATEGORY];
        }

        public function get_title()
        {
            return 'Featured image (Custom Size)';
        }

        public function get_value(array $options = [])
        {
            $thumbnail_id = get_post_thumbnail_id();

            if ($thumbnail_id) {
                $size = $this->get_settings('size');
                $image_data = [
                    'id' => $thumbnail_id,
                    'url' => wp_get_attachment_image_src($thumbnail_id, $size)[0],
                ];
            } else {
                $image_data = $this->get_settings('fallback');
            }

            return $image_data;
        }

        protected function _register_controls()
        {
            $this->add_control(
                'size',
                [
                    'label' => __('Size', 'elementor-pro'),
                    'type' => Controls_Manager::TEXT,
                ]
            );

            $this->add_control(
                'fallback',
                [
                    'label' => __('Fallback', 'elementor-pro'),
                    'type' => Controls_Manager::MEDIA,
                ]
            );
        }
    }

    $dynamic_tags->register_tag('Featured_Image_Custom_Size');
});```

**Comment by biocas** [comment]
+1

**Comment by wwdes** [comment]
+1

**Comment by walik-walik** [comment]
+1

**Comment by gkarmas** [comment]
Come on guys this is like a super basic feature. Why are you even limiting the image size to a low res selection?

**Comment by webenadam** [comment]
3 Years later...

**Comment by ilyakar** [comment]
+1. This isn't a want, it's a need.

**Comment by R2wix9mtdI** [comment]
> 
> 
> +1. This isn't a want, it's a need.

Why do I thik this is falling on deaf ears. I loved the Elementor CMS but have been forced to use the native Gutenberg editor. The more they ignore these replies, the more people are going to look for alteratives. I'm now using Astra and Gutenberg as the editor. It's focus is light weight and speed - get this right first team and I'll switch back, but in the meantime...

**Comment by julian-stark** [comment]
+1

**Comment by aureliePrd** [comment]
+1

**Comment by hmouhtar** [comment]
+1

**Comment by garrulinae** [comment]
To everyone commenting on this old closed request, a reminder to show your support on the active request - #6778. Go there and add a thumbs up and comment.

**Comment by ConsultMePL** [comment]
1/11/2021

3 years, 7 months and it's still not here? 

I have changed the why im working with pictures, as design wise quite often it works way better to set a picture as section or background image. Although my site after caching loads pretty much immediately I was surprised why the hell GT Metrix score is so low - then it turned, it's because background images can't be resized...

**Comment by marcieee73** [comment]
+1
:(

**Comment by youssefhamdane-cl** [comment]
+1

**Comment by Jonatan-r** [comment]
Very much needed still. Why is this closed? Why is this not replied by to by elementor? @elementor

**Comment by Xavvi** [comment]
+1

**Comment by periwinkleFr** [comment]
++++++1
It could be really appreciated when using in loop templates.

**Comment by speedxcz** [comment]
Hi, the question has been open since 2018, it would be worth solving ;) +1

**Comment by d7omepro-vr** [comment]
Hello everyone...
Today I reported this to elementor and I got a pretty good solution

`selector .elementor-widget-wrap {
background-size: 100% 100% !important;
}`

I hope this useful

**Comment by warudin** [comment]
@d7omepro-vr the problem is that Elementor always uses the original uploaded image, while in most situations it could use a smaller thumbnail.

This can result in an extra pageload of several MB's per case and tens of MB's per page if it's being used in a loop.

**Comment by wzzly** [comment]
+1 here! This is a major feature that should be added in my eyes.

**Comment by MathijsEnter** [comment]
+1. The original size is not always required for a background, but rather a scaled version  of it. It is also not always possible to resize it within WP, as an image can be used multiple times. Having to upload the image twice just so it can be scaled to use it as a background is counterintuitive.

**Comment by christianmagill** [comment]
@nicholaszein, I'm glad to see this merged into beta! Any rough estimate on when it might hit full release?

**Comment by nicholaszein** [comment]
Hello everyone!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor **v3.11**! 🥳 

✔️ Feel free to check it out and update your plugin to the new version!

Cheers 🥂
